include/spdlog/sinks/daily_file_sink.h

```diff
@@ -34,6 +39,98 @@ struct daily_filename_calculator
+                case 'b':
+                    dest += spdlog::details::months[static_cast<size_t>(now_tm.tm_mon)];
+                    break;
+                case 'B':
+                    dest += spdlog::details::full_months[static_cast<size_t>(now_tm.tm_mon)];
+                    break;
+                case 'c':
+                    dest += calc_filename_fmt("%a %b %d %H:%M:%S %Y", now_tm);
+                    break;
+                case 'C':
+                    dest += fmt::format ("{}", (now_tm.tm_year + 1900) % 100);
+                    break;
+                case 'Y':
+                    dest += fmt::format ("{}", now_tm.tm_year + 1900);
+                    break;
+                case 'D':
+                case 'x':
+                    dest += calc_filename_fmt("%m %d %C", now_tm);
+                    break;
+                case 'm':
+                    dest += fmt::format ("{:02}", now_tm.tm_mon + 1);
+                    break;
+                case 'd':
+                    dest += fmt::format ("{:02}", now_tm.tm_mday);
+                    break;
+                case 'H':
+                    dest += fmt::format ("{:02}", now_tm.tm_hour);
+                    break;
+                case 'I':
+                    dest += fmt::format ("{:02}", (now_tm.tm_hour % 12) == 0 ? 12 : (now_tm.tm_hour % 12));
+                    break;
+                case 'M':
+                    dest += fmt::format ("{:02}", now_tm.tm_min);
+                    break;
+                case 'S':
+                    dest += fmt::format ("{:02}", now_tm.tm_sec);
+                    break;
+                case 'p':
+                    dest += (now_tm.tm_hour >= 1 && now_tm.tm_hour <= 12 ? "AM" : "PM");
+                    break;
+                case 'r':
+                    dest += calc_filename_fmt("%I:%M:%S %p", now_tm);
+                    break;
+                case 'R':
+                    dest += calc_filename_fmt("%H:%M", now_tm);
+                    break;
+                case 'T':
+                case 'X':
+                    dest += calc_filename_fmt("%H:%M:%S", now_tm);
+                    break;
+                case 'z':
+                    total_minutes = spdlog::details::os::utc_minutes_offset ();
+                    if (total_minutes >= 0) {
+                        dest += fmt::format ("+{:02}:{:02}", total_minutes / 60, total_minutes % 60);
+                    } else {
+                        total_minutes = -total_minutes;
+                        dest += fmt::format ("-{:02}:{:02}", total_minutes / 60, total_minutes % 60);
+                    }
+                    break;
+                case '%':
+                    dest += '%';
+                    break;
+                default:
+                    throw_spdlog_ex("daily_file_sink: Invalid format");
+                }
+            } else {
+                dest += filename [i];
+            }
+        }
+        return dest;
```

Comment:
why not replace all this and  receive param of  fmt chrono format string? this would make the impl trivial (https://fmt.dev/dev/api.html#chrono-api)